src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedPrimaryConstructor.cs

```diff
@@ -28,9 +28,15 @@ internal sealed class SynthesizedPrimaryConstructor : SourceConstructorSymbolBas
 
             this.MakeFlags(
                 MethodKind.Constructor,
+                RefKind.None,
                 containingType.IsAbstract ? DeclarationModifiers.Protected : DeclarationModifiers.Public,
                 returnsVoid: true,
+                // We consider synthesized constructors to have a body, since they effectively span the entire type, and
+                // can do things like create constructor assignments that write into the fields/props of the type.
+                hasAnyBody: true,
+                isExpressionBodied: false,
                 isExtensionMethod: false,
+                isVarArg: syntax.ParameterList?.Parameters.Any(static p => p.IsArgList) ?? false,
```

Comment:
>?. [](http://example.com/codeflow?start=46&length=2)

I think conditional access is unnecessary here. We wouldn't have a primary constructor if there was no a parameter list. To suppress a nullability warning, we can simply assert the fact that ```syntax.ParameterList is not null```. See us doing the same in ```TryGetBodyBinder```, and using ```!``` in ```GetParameterList```.

Reply:
Added assert.  Thanks :)